Release Note Sprint 09-10 (#35)
mia-platform/documentation: docs/release-notes/platform_10-6-0_releasenotes.md

:::info This feature is still in BETA, it is under active development. :::

Are you tired of writing ER Schemas manually? The new Fast Data ER Schema No Code Configuration is now generally available!
 
It is now possible to create and generate relationships between Projections using a simple and easy to use No Code configuration.  
This tool is able to generate a JSON that will be used by the Single View Creator. The generated JSON is exactly the same JSON that would have been written manually in the Low Code Configuration.

In order to exploit this functionality you have to open the Projections section and navigate in the new ER Schema tab. You can create a new ER Schema and start designing it!   
When configuring the Single View Creator, in the Single Views section, you just have to select one of the ER Schemas created in the aforementioned section from a dropdown list.

This feature will only be available for PaaS users and it can be activated only for specific projects. If you want to try it out, contact your system administrator!
Do not miss out the official documentation here! (<TODO add documentation link>)
(<TODO add screenshot>)

> #### Feature Preview section (only available for PaaS users)

Reviewer: ```suggestion
#### Feature Preview section (only available for SaaS Console users)
```

> PaaS users can now have visibility of features in Console that are still in preview and decide to try them out and give us precious feedback.  

Reviewer: ```suggestion
SaaS users can now have visibility of features in Console that are still in preview and decide to try them out and give us precious feedback.  
```

---
id: v10.6.0
title: Version 10.6.0 Release Notes
sidebar_label: v10.6.0
image: "img/release-note-link-preview.png"
---

_March 16, 2023_

## Console

### New Features

> In this Console release we have brought exciting New Features which are currently available exclusively for PaaS users. 

Reviewer 2: ```suggestion
In this Console release we have brought exciting New Features which are currently available exclusively for SaaS users. 
```

> Here below you can discover all of them!

Reviewer 2: da rimuovere

> Here below you can discover all of them!

Reviewer 2: should be removed

> #### Fast Data No Code Configuration for ER Schema (only available for PaaS users)

Reviewer 2: ```suggestion
#### Fast Data No Code Configuration for ER Schema (available for SaaS)
```

:::info This feature is still in BETA, it is under active development. :::

Are you tired of writing ER Schemas manually? The new Fast Data ER Schema No Code Configuration is now generally available!
 
It is now possible to create and generate relationships between Projections using a simple and easy to use No Code configuration.  
This tool is able to generate a JSON that will be used by the Single View Creator. The generated JSON is exactly the same JSON that would have been written manually in the Low Code Configuration.

In order to exploit this functionality you have to open the Projections section and navigate in the new ER Schema tab. You can create a new ER Schema and start designing it!   
When configuring the Single View Creator, in the Single Views section, you just have to select one of the ER Schemas created in the aforementioned section from a dropdown list.

> This feature will only be available for PaaS users and it can be activated only for specific projects. If you want to try it out, contact your system administrator!

Reviewer 2: ```suggestion
This feature is available only for SaaS users and it can be activated only for specific projects. If you want to try it out, contact your system administrator!
```

Do not miss out the [official documentation](/fast_data/configuration/config_maps/erSchema.md#use_the_no_code)! 

![Fast Data No Code](./img/10.6/fast_data_no_code.png)

#### Feature Preview section (only available for SaaS Console users)

SaaS users can now have visibility of features in Console that are still in preview and decide to try them out and give us precious feedback.  

> For instance, once entered in this section, PaaS users can now try the new `Dynamic Sidebar` of the Design Area.   

Reviewer 2: ```suggestion
For instance, once entered in this section, SaaS users can now try the new `Dynamic Sidebar` of the Design Area.   
```

Author: why is it "Saas" and not "Paas"? We have always referred to the cloud Console as the "Paas Console", I think

Reviewer 2: That's because these new features are specific to the Console itself (which is SaaS). The infrastructure & runtime offered by the PaaS solution are not taken into account here.

To discover more, just try this new feature and tell us what you think!
 
This section can be accessed from the Launchbar by clicking on the `Feature Preview` item inside the dropdown menu that appears when hovering over the user avatar. 

![Feature Preview Modal - Feature Preview modal with some features enabled](./img/10.6/feature-preview.png)

> #### User Preferences section (only available for PaaS users)

Reviewer: This section will only be published if we get the updated privacy policy for email notifications; if we don't get the privacy policy we won't release neither the email notification feature nor this feature (which currently is only meaningful if the email notification feature is active)

> #### User Preferences section (only available for PaaS users)

Reviewer 2: ```suggestion
#### User Preferences section (only available for SaaS users)
```

> We have introduced a new section named `User Preferences`, in which PaaS users can customize their experience in Console according to their own preferences.  

Reviewer 2: ```suggestion
We have introduced a new section named `User Preferences`, in which SaaS users can customize their experience in Console according to their own preferences.  
```

#### PATCH & PUT support in http actions
Actions of type `http` support REST methods `PATCH` and `PUT`

### Bug Fix

#### `bk-layout-container` forwards headers
Component `bk-layout-container` forwards property `headers` to its content

#### `bk-layout-container` injects headers
Component `bk-layout` adds headers as specified in property `headers` to its http-calls

#### Locales are correctly loaded
Locales are correctly loaded on first plugin render

#### File upload action notifies http-call fail
Actions of type `file-upload` calls `onError` hook if http-call fails

#### Href actions resolve dynamic configurations
Actions of type `href` correctly resolve dynamic configurations

#### `bk-layout-container` does not make unnecessary http-calls
Component `bk-layout-container` does not trigger extra http-calls

#### Fields keep focus in wizard
Fields are correctly put on focus on `bk-form-modal` wizard mode

#### `__STATE__` field is not discarded
Form components do not discard `__STATE__` field on data creation

### Search bar keeps state
Component `bk-search-bar` keeps state on layout change when placed inside `bk-layout-container` component

## How to update your Console

> For on-premise Console installations, please contact your Mia Platform referent to know how to use the `Helm chart version X.X.X`.

Reviewer 2: Helm chart version to be inserted

Reviewer 3: 9.0.0 version added
